e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/procedural/scheduling/DatabaseDeletionTests.java

```diff
@@ -0,0 +1,177 @@
+
+import gov.nasa.jpl.aerie.e2e.types.GoalInvocationId;
+import gov.nasa.jpl.aerie.e2e.utils.GatewayRequests;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import javax.json.Json;
+import javax.json.JsonValue;
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class DatabaseDeletionTests extends ProceduralSchedulingSetup {
+  private GoalInvocationId procedureId;
+
+  @BeforeEach
+  void localBeforeEach() throws IOException {
+    try (final var gateway = new GatewayRequests(playwright)) {
+      int procedureJarId = gateway.uploadJarFile("build/libs/DeleteBiteBananaGoal.jar");
+      // Add Scheduling Procedure
+      procedureId = hasura.createSchedulingSpecProcedure(
+          "Test Scheduling Procedure",
+          procedureJarId,
+          specId,
+          0
+      );
+    }
+  }
+
+  @AfterEach
+  void localAfterEach() throws IOException {
+    hasura.deleteSchedulingGoal(procedureId.goalId());
+  }
+
+  @Test
+  void deletesDirectiveAlreadyInDatabase() throws IOException {
+    hasura.insertActivityDirective(
+        planId,
+        "BiteBanana",
+        "1h",
+        JsonValue.EMPTY_JSON_OBJECT
+    );
+
+    var plan = hasura.getPlan(planId);
+    assertEquals(1, plan.activityDirectives().size());
+
+    hasura.awaitScheduling(specId);
+
+    plan = hasura.getPlan(planId);
+    assertEquals(0, plan.activityDirectives().size());
+  }
+
+  @Test
+  void deletesDirectiveInDatabaseWithAnchor() throws IOException {
+    final var bite = hasura.insertActivityDirective(
+        planId,
+        "BiteBanana",
+        "1h",
+        JsonValue.EMPTY_JSON_OBJECT
+    );
+
+    final var grow = hasura.insertActivityDirective(
+        planId,
+        "GrowBanana",
+        "1h",
+        JsonValue.EMPTY_JSON_OBJECT,
+        Json.createObjectBuilder().add("anchor_id", bite)
+    );
+
+    var plan = hasura.getPlan(planId);
+    var activities = plan.activityDirectives();
+    assertEquals(2, activities.size());
+    assertTrue(activities.stream().anyMatch(
+        it -> Objects.equals(it.type(), "GrowBanana")
+              && Objects.equals(it.anchorId(), bite)
+              && Objects.equals(it.startOffset(), "01:00:00")
+    ));
+
+    hasura.awaitScheduling(specId);
+
+    plan = hasura.getPlan(planId);
+
+    activities = plan.activityDirectives();
+    assertEquals(1, activities.size());
+
+    assertTrue(activities.stream().anyMatch(
+        it -> Objects.equals(it.type(), "GrowBanana")
+        && Objects.equals(it.anchorId(), null)
+        && Objects.equals(it.startOffset(), "02:00:00")
+    ));
+  }
+
+  @Test
+  void deletesDirectiveInDatabaseInMiddleOfChain() throws IOException {
+
+    // Creates 5 activities, deletes "Bite".
+    // grow1 <- bite
+    // bite <- grow (id lost)
```

Comment:
What does `id lost` mean here?

Reply:
It just means I didn't record it in a variable, because I didn't anchor anything to it.